lib/krb5/expand_path.c

```diff
@@ -349,6 +359,19 @@ _expand_loginname(krb5_context context, PTYPE param, const char *postfix, char *
     return 0;
 }
 
+static krb5_error_code
+_expand_strftime(krb5_context context, PTYPE param, const char *postfix,
+		 const char *arg, char **ret)
+{
+    time_t t;
+    char buf[1024];
+
+    t = time(NULL);
+    strftime(buf, sizeof(buf), arg, localtime(&t));
```

Comment:
The return value of strftime should probably be checked.

Reply:
Well, that raises an interesting point. strftime(3) returns only a length, there are no errors. I was operating under the assumption that it was acceptable to accept a truncated result and therefore we do not need to check the error because, IIRC, strftime will nul-terminate the truncated string. We could, however, log an error and bail---but that raises the question: where would the log go if we can't open the log file? So, there's a few choices here.

Reply:
The FreeBSD manpage suggests that strftime returns zero on error, and then buffer is indeterminate. There's never truncation.  If we get 0, then we can't use the buffer...
```
     No more than maxsize characters will be placed into the array.  If the
     total number of resulting characters, including the terminating NUL
     character, is not more than maxsize, strftime() returns the number of
     characters in the array, not counting the terminating NUL.  Otherwise,
     zero is returned and the buffer contents are indeterminate.
```
I would naively also expect zero for invalid conversion specifiers.

Reply:
But a test with `date +"%f%J%i"` shows that invalid conversions just emit the letter in question and drop the `%`, so I get `fJi` for the above...